net/modemmanager/files/modemmanager.init

```diff
@@ -5,32 +5,24 @@ USE_PROCD=1
 START=70
 
 stop_service() {
-	# Load common utils
 	. /usr/share/ModemManager/modemmanager.common
-	# Set all configured interfaces as unavailable
 	mm_cleanup_interfaces
 }
 
-start_service() {
-	# Setup ModemManager service
-	#
-	# We will make sure that the rundir always exists, and we initially cleanup
-	# all interfaces flagging them as unavailable.
-	#
-	# The cached events processing will wait for MM to be available in DBus
-	# and will make sure all ports are re-notified to ModemManager every time
-	# it starts.
-	#
-	# All these commands need to be executed on every MM start, even after
-	# procd-triggered respawns, which is why they're all included as instance command
-	#
+start_mm() {
+	. /usr/share/ModemManager/modemmanager.common
```

Comment:
This is more or less what we had before, with the daemon call as only command in the procd instance. But having it that way ended up failing because mm_report_events_from_cache was not being called if the service was restarted by procd itself e.g. after a daemon crash (as explained in the comment that was also removed: "All these commands need to be executed on every MM start, even after procd-triggered respawns, which is why they're all included as instance command").

How are these changes avoiding that? If these changes prevent that problem, I'm all for them, but I'm not really sure if they do. Isn't procd supposed to re-run the "command" configured only on daemon crashes?

Reply:
If that's the case then I'd rather put it all in a wrapper/startup script `mm-service`?
Then it looks better right away.
We also could fix the two issues as mentioned (`ubus service` and `pdifle`)

Reply:
Yes, that is probably the cleanest way forward

Reply:
@aleksander0m I have now outsourced the startup sequence to a wrapper script.